### PR TITLE
Combine and transfer older work to new repo

### PR DESCRIPTION
**Pull request by MichaelClifford**
There has been a few disparate efforts developing with Ray both from a users perspective and an OpenShift deployment perspective. With the start of our new Joint OKR we wanted to bring everything together under one Red Hat ET repo we could move forward from. So this PR simple adds the relevant work to date from a couple different repos into this new one.

The additions here consist of work done by @MichaelClifford @erikerlandson @harshad16

### CODE REVIEW

**Comment by MichaelClifford** [comment]
For reference, the places this has been compiled from are: 
* https://github.com/MichaelClifford/moc-ray-demo
* https://github.com/erikerlandson/moc-ray-demo 
* https://github.com/opendatahub-io/odh-manifests/pull/573

**Review by erikerlandson — APPROVED** [review]

**Comment by erikerlandson** [comment]
lgtm, thanks!

File: notebooks/ray-examples/ray-datasets.ipynb [hunk 1]
```diff
@@ -0,0 +1,1961 @@
...
+   "source": [
+    "# Ray Data Test Notebook\n",
+    "\n",
+    "The vast majority of this notebook is based off implementing the examples made available by the ray data [getting started](https://docs.ray.io/en/latest/data/getting-started.html#datasets-getting-started) docs. \n",
+    "\n",
+    "\n",
+    "\n",
+    "### What kinds of things should I use Ray Datasets for? \n",
+    "\n",
+    "According to their docs, Ray is, \"designed to load and pre-process data for distributed ML training pipelines...Ray Datasets is not intended as a replacement for more general data processing systems\"[[1]]. Its purpose is only to serve as a \"last mile\" distributed data processing tool. Therefore it is designed with the following 3 use cases in mind. \n",
```
Comment (Shreyanand) [inline notebooks/ray-examples/ray-datasets.ipynb:15]: Do we have a position for a general ETL tool? It would be cool to see Ray working with an ETL tool for an end to end ML project.

File: notebooks/ray-examples/ray-datasets.ipynb [hunk 2]
```diff
@@ -0,0 +1,1961 @@
...
+   "source": [
+    "Above we have evaluated 3 scenarios: Using vanilla pandas with a single dataset, using Ray with a single dataset and using Ray with a distributed dataset for a number of different operations. \n",
+    "\n",
+    "Our generated datasets are 100,000,000 rows long and 1 column wide, consisting only of integers. Below we have recorded the timing and memory results for loading the data, retrieving a slice (subset), applying the square function to each element and applying a filter to the dataset along with the total time taken to perform each step and the memory still in use after the entire set of operations ran. \n",
+    "\n",
+    "#### Ray vs Pandas performance results\n",
+    "\n",
+    "_These are the experimental results on an 8 core laptop with 32Gb Memory and should be repeated on an OPF cluster._ \n",
+    "\n",
+    "_TODO: Redo assessment once we get it working on the cluster._\n",
```
Comment (Shreyanand) [inline notebooks/ray-examples/ray-datasets.ipynb:717]: are the local machine comments still required?

File: notebooks/ray-examples/ray-torch.ipynb [hunk 3]
```diff
@@ -0,0 +1,455 @@
+{
+ "cells": [
```
Comment (Shreyanand) [inline notebooks/ray-examples/ray-torch.ipynb:2]: pytorch usually has a dataloader class for creating batches and padding etc for the Neural network, I'm curious how that would play out with ray datasets...

File: notebooks/ray-examples/ray-tune.ipynb [hunk 4]
```diff
@@ -0,0 +1,1375 @@
+{
```
Comment (Shreyanand) [inline notebooks/ray-examples/ray-tune.ipynb:1]: This notebook also shows using ray with sklearn which seems to be a good use case. Would renaming the notebook or adding markdown to highlight that make sense?

**Review by Shreyanand — COMMENTED** [review]
Just saw this got merged :D  I just went through the example notebooks, adding some minor comments for later.
